# Peer-graded Assignment: Running Your First Program

Submit by February 4, 11:59 PM PST

**Important Information**

It is especially important to submit this assignment before the deadline, February 4, 11:59 PM PST, because it must be graded by others. If you submit late, there may not be enough classmates around to review your work. This makes it difficult - and in some cases, impossible - to produce a grade. Submit on time to avoid these risks.

## Instructions
    
This week, you learned how to call in a dataset, run frequency distributions for your chosen variables and select the columns (i.e. variables), and possibly rows (i.e. observations). This week’s assignment will give you the opportunity to display what you’ve learned as you run your first program and start focusing on your selected data set in relation to your research question.

## Review criteria

Your assessment will be based on the evidence you provide that you have completed all of the steps. When relevant, gradients in the scoring will be available to reward clarity (for example, you will get one point for submitting output that is not understandable, but two points if it is understandable). In all cases, consider that the peer assessing your work is likely not an expert in the field you are analyzing. You will be assessed equally on your description of your frequency distributions.

Specific rubric items, and their point values, are as follows:

* Was the program output interpretable (i.e. organized and labeled)? (1 point)
* Does the program output display three data managed variables as frequency tables? (1 point)
* Did the summary describe the frequency distributions in terms of the values the variables take, how often they take them, the presence of missing data, etc.? (2 points)

## Instructions

Continue with the dataset you’ve been working with.

* STEP 1: Run your first program. This program will be used throughout the remainder of the course and become the basis of your data analysis going forward.

Be sure to correct any errors highlighted in the program.

* STEP 2: Run frequency distributions for your chosen variables and select columns, and possibly rows.

Your output should be interpretable (i.e. organized and labeled).

## WHAT TO SUBMIT:

Following completion of your first program, create a blog entry where you post 
1. your program 
2. the output that displays three of your variables as frequency tables and 
3. a few sentences describing your frequency distributions in terms of the values the variables take, how often they take them, the presence of missing data, etc.

In [24]:
import pandas as pd
import numpy as np

# Explore the number of male and female population

There is also a third gender P for Person. I would like to assume P = F + M. But we will see if this is the case.

In [25]:
vic = pd.read_csv('2016_census_vic_income.csv')

In [26]:
vic['gender'].value_counts()

M    94230
F    94230
P    94230
Name: gender, dtype: int64

In [27]:
# Male population
pop_m = vic['number'][vic['gender'] == 'M'].sum()

In [28]:
# Female population
pop_f = vic['number'][vic['gender'] == 'F'].sum()

In [29]:
# Person population
pop_p = vic['number'][vic['gender'] == 'P'].sum()

In [30]:
print(f'Male population is {pop_m}')
print(f'Female population is {pop_f}')
print(f'Person population is {pop_p}')
print('Person populatin is equal to Male + Female population?', pop_p == pop_m + pop_f)
print(f'Person population: {pop_p}.\tM+F population: {pop_m+pop_f}')

Male population is 2339649
Female population is 2480205
Person population is 4832708
Person populatin is equal to Male + Female population? False
Person population: 4832708.	M+F population: 4819854


Because the Person (total) population doesn't not add up to the male plus female population, this shows that we can't assume there are only two genders in the data.

# Explore the variables

* age range
* income brackets
* gender (we will only compare females and males, ignoring persons at the moment)

In [31]:
vic.head()

postcode gender  age_low  age_high  income_low  income_high  number
0  POA3000      M       15        19         1.0        149.0      78
1  POA3002      M       15        19         1.0        149.0       6
2  POA3003      M       15        19         1.0        149.0      17
3  POA3004      M       15        19         1.0        149.0      10
4  POA3005      M       15        19         1.0        149.0       0

In [32]:
# age range

age_range = pd.unique(vic['age_low'].astype('int')), pd.unique(vic['age_high'].astype('int'))
age_range

(array([15, 20, 25, 35, 45, 55, 65, 75, 85], dtype=int64),
 array([19, 24, 34, 44, 54, 64, 74, 84, 99], dtype=int64))

In [33]:
# double check there is no Non/NaN value
sum(vic['income_low'].isnull().values), sum(vic['income_high'].isnull().values)

(0, 0)

In [34]:
# income brackets
#  -99 is None (unknown)
# -1 is for negative income

pd.unique(vic['income_low'].astype('int'))

array([   1,  150,  300,  400,  500,  650,  800, 1000, 1250, 1500, 1750,
       2000, 3000,  -99,   -1], dtype=int64)

In [35]:
pd.unique(vic['income_high'].astype('int'))

array([ 149,  299,  399,  499,  649,  799,  999, 1249, 1499, 1749, 1999,
       2999, 9999,  -99,    0], dtype=int64)

In [36]:
# There is no guarantee the income range will appear in pair in the dataframe.
# It will be needed to find a way to verify that.

income_brackets = list(zip(
    pd.unique(vic['income_low'].astype('int')),
    pd.unique(vic['income_high'].astype('int')),
))
income_brackets

[(1, 149),
 (150, 299),
 (300, 399),
 (400, 499),
 (500, 649),
 (650, 799),
 (800, 999),
 (1000, 1249),
 (1250, 1499),
 (1500, 1749),
 (1750, 1999),
 (2000, 2999),
 (3000, 9999),
 (-99, -99),
 (-1, 0)]

### Frequency table of:
* Total male/female population
* Total male/female population has 0, negative or unknown income
* Male/female population with a positive income
* Frequency tables of male/female based by age

In [37]:
# number of rows, not the population
vic['gender'].value_counts()


M    94230
F    94230
P    94230
Name: gender, dtype: int64

In [38]:
vic_f = vic[vic['gender'] == 'F']
vic_f_pop= sum(vic_f['number'])
vic_m = vic[vic['gender'] == 'M']
vic_m_pop = sum(vic_m['number'])
# vic_total_population = vic_female_total_population + vic_male_total_population
vic_p = vic[vic['gender'] == 'P']
vic_p_pop = sum(vic_p['number'])

In [39]:
print(f'Female Population in Victoria is {vic_f_pop}. {vic_f_pop/vic_p_pop*100:.4f}%')
print(f'Male Population in Victoria is {vic_m_pop}. {vic_m_pop/vic_p_pop*100:.4f}%')

Female Population in Victoria is 2480205. 51.3212%
Male Population in Victoria is 2339649. 48.4128%


In [40]:
vic_f_pos_income_pop = vic_f[vic_f['income_low'] >= 1]['number']

In [41]:
# Positive income groups
vic_f_pos_income = vic_f[vic_f['income_low'] >= 1]
vic_f_pos_income_pop = vic_f_pos_income['number'].sum()
vic_m_pos_income = vic_m[vic_m['income_low'] >= 1]
vic_m_pos_income_pop = vic_m_pos_income['number'].sum()
vic_f_pos_income_pop, vic_m_pos_income_pop


(1972735, 1933300)

In [42]:
print(f'Female Population that has a positive income in Victoria is {vic_f_pos_income_pop}. {vic_f_pos_income_pop/vic_p_pop*100:.4f}%')
print(f'Male Population  that has a positive income in Victoria is {vic_m_pos_income_pop}. {vic_m_pos_income_pop/vic_p_pop*100:.4f}%')

Female Population that has a positive income in Victoria is 1972735. 40.8205%
Male Population  that has a positive income in Victoria is 1933300. 40.0045%


In [43]:
#vic_f_pos_income
dfv = pd.DataFrame()
dfv['f positive income pop'] = vic_f_pos_income.groupby(['income_low'])['number'].sum()
dfv['m positive income pop'] = vic_m_pos_income.groupby(['income_low'])['number'].sum()
dfv['f positive income percentage'] = vic_f_pos_income.groupby(['income_low'])['number'].sum() / vic_p_pop * 100
dfv['m positive income percentage'] = vic_m_pos_income.groupby(['income_low'])['number'].sum() / vic_p_pop * 100

# Weekly Minimum Wages of Females and Males in Victoria
## Variable One: Gender
If you use the minimum wages as a reference point, you can see that below the minimum wages, females has more percentages over males. That is \$650 group, where females has 4.115477% and males 3.435548%.
But above the minimum wages, it's the other way around and the gap grows wider when the income goes higher.


## Minimum Wages
The national minimum wage is currently \$18.29 per hour or \$694.90 per 38 hour week (before tax). [Source](https://www.fairwork.gov.au/how-we-will-help/templates-and-guides/fact-sheets/minimum-workplace-entitlements/minimum-wages#)

## Frequency table:

In [44]:
dfv

f positive income pop  m positive income pop  \
income_low                                                 
1.0                        133977                  82802   
150.0                      206825                 140845   
300.0                      235589                 169565   
400.0                      241210                 148428   
500.0                      216936                 142528   
650.0                      198889                 166030   
800.0                      193741                 209125   
1000.0                     181191                 224274   
1250.0                     118557                 157712   
1500.0                      89988                 135111   
1750.0                      54054                  97569   
2000.0                      65081                 153912   
3000.0                      36697                 105399   

            f positive income percentage  m positive income percentage  
income_low                                                              
1.0                             2.772297                      1.713367  
150.0                           4.279692                      2.914412  
300.0                           4.874886                      3.508695  
400.0                           4.991197                      3.071322  
500.0                           4.488912                      2.949237  
650.0                           4.115477                      3.435548  
800.0                           4.008953                      4.327284  
1000.0                          3.749264                      4.640752  
1250.0                          2.453221                      3.263429  
1500.0                          1.862062                      2.795762  
1750.0                          1.118503                      2.018930  
2000.0                          1.346678                      3.184798  
3000.0                          0.759347                      2.180951

The national minimum wage is currently \$18.29 per hour or \$694.90 per 38 hour week (before tax). 

# Inspect four special groups based on the post code
## Variable Two: Post Code
1. people who have zero or negative income
2. people who have unknown income
3. people who have about the minimum wage income
4. people who have the highest income


In [49]:
vic_p = vic[vic['gender'] == 'P']
vic_p.head()

postcode gender  age_low  age_high  income_low  income_high  number
146580  POA3000      P       15        19         1.0        149.0     221
146581  POA3002      P       15        19         1.0        149.0       9
146582  POA3003      P       15        19         1.0        149.0      32
146583  POA3004      P       15        19         1.0        149.0      30
146584  POA3005      P       15        19         1.0        149.0       3

You can see nine age groups of negative or zero income population in each post code.
We will combine them into one postcode group

In [133]:
vic_p[(vic_p['postcode'] == 'POA3000') & (vic_p['income_low'] == -1)]

postcode gender  age_low  age_high  income_low  income_high  number
270824  POA3000      P       15        19        -1.0          0.0    1981
271522  POA3000      P       20        24        -1.0          0.0    5254
272220  POA3000      P       25        34        -1.0          0.0    2182
272918  POA3000      P       35        44        -1.0          0.0     288
273616  POA3000      P       45        54        -1.0          0.0     183
274314  POA3000      P       55        64        -1.0          0.0     132
275012  POA3000      P       65        74        -1.0          0.0      58
275710  POA3000      P       75        84        -1.0          0.0      16
276408  POA3000      P       85        99        -1.0          0.0       6

In [132]:
# sanity check
vic_p[(vic_p['postcode'] == 'POA3000') & (vic_p['income_low'] == -1)]['number'].sum()

10100

In [134]:
vic_pp = vic_p.groupby(['postcode', 'income_low'])['number'].sum()
vic_pp.head(20)

postcode  income_low
POA3000   -99.0          4678
          -1.0          10100
           1.0           1256
           150.0         1868
           300.0         2168
           400.0         2052
           500.0         2162
           650.0         1910
           800.0         2096
           1000.0        2227
           1250.0        1619
           1500.0        1322
           1750.0         884
           2000.0        1483
           3000.0         923
POA3002   -99.0           602
          -1.0            245
           1.0             84
           150.0          161
           300.0          128
Name: number, dtype: int64

In [154]:
# try to get a value out of a multi-indexed data frame
vic_pp.xs(-1, level='income_low').head()

postcode
POA3000    10100
POA3002      245
POA3003      615
POA3004      802
POA3005       32
Name: number, dtype: int64

In [173]:
# Postcode average population
# Don't use this. Use the whole population. (vic_p_pop)
print('Victoria Population:', vic_p_pop)
postal_pop_avg = vic.groupby('postcode')['number'].mean()
postal_pop_avg.head()

Victoria Population: 4832708


postcode
POA3000    181.540741
POA3002     22.837037
POA3003     25.083951
POA3004     43.128395
POA3005      2.187654
Name: number, dtype: float64

In [157]:

vic_p_negzero = vic_pp.xs(-1, level='income_low')
vic_p_unknown = vic_pp.xs(-99, level='income_low')
vic_p_minimum = vic_pp.xs(650, level='income_low')
vic_p_highest = vic_pp.xs(3000, level='income_low')
vic_p_4incomes_num = pd.concat([vic_p_negzero, vic_p_unknown, vic_p_minimum, vic_p_highest], axis=1, 
                          keys=['neg or zero', 'unknown', 'minimum', 'highest'])

vic_p_4incomes_percent = vic_p_4incomes_num / vic_p_pop * 100

## Frequency table: incomes of four types (count)

In [162]:
vic_p_4incomes_num

neg or zero  unknown  minimum  highest
postcode                                        
POA3000         10100     4678     1910      923
POA3002           245      602      202      609
POA3003           615      861      281      224
POA3004           802     1241      399      785
POA3005            32      133       18       31
POA3006          2494     2456      915     1149
POA3008          1656     1231      452      638
POA3010           452       72       21       15
POA3011          1784     2079     1196      407
POA3012          2220     2017     1540      486
POA3013           890      890      695      713
POA3015          1095      985      775      840
POA3016          1223     1024      660     1143
POA3018           883      891      688      327
POA3019          1138      633      621       58
POA3020          4764     3540     2987      211
POA3021          6375     4461     3727      183
POA3022           319      279      204       14
POA3023          6762     4736     3827      526
POA3024          2151     1325     1502      147
POA3025           972     1053      638      139
POA3026             0       15        3        0
POA3027           761      296      339      137
POA3028          2727     1857     1800      210
POA3029          9857     6010     5913      642
POA3030          9740     5686     5551     1746
POA3031          1152     1623      836      727
POA3032          2450     2083     1687     1153
POA3033          1150      995      810      274
POA3034           999      873      650      189
...               ...      ...      ...      ...
POA3950           249      352      367       48
POA3951            68       84       64       12
POA3953           449      626      519       88
POA3954            30       16       40        0
POA3956           171      417      180       37
POA3957            17       20       21        3
POA3958            18       27       13        3
POA3959            61      106       82       16
POA3960           133      269      146       23
POA3962            52       80       67       11
POA3964            11        0       13        0
POA3965            18       35       12        0
POA3966            40       26       49        6
POA3967             3       12        6        0
POA3971           248      383      274       47
POA3975          1570      879      954      112
POA3976          2983     1724     1963       69
POA3977          7994     5650     6665      713
POA3978           850      550      725      109
POA3979            19       23       30        0
POA3980           211      252      224       29
POA3981           326      469      375       46
POA3984           315      483      371       49
POA3987           119      120       94       26
POA3988            83       76       86       11
POA3990             5        9       14        0
POA3991            25       30       19        3
POA3992            60       61       82       12
POA3995           496      975      772       93
POA3996           259      449      390      108

[698 rows x 4 columns]

## Frequency table: incomes of four types (percentage)

In [164]:
vic_p_4incomes_percent


neg or zero   unknown   minimum   highest
postcode                                           
POA3000      0.208993  0.096799  0.039522  0.019099
POA3002      0.005070  0.012457  0.004180  0.012602
POA3003      0.012726  0.017816  0.005815  0.004635
POA3004      0.016595  0.025679  0.008256  0.016243
POA3005      0.000662  0.002752  0.000372  0.000641
POA3006      0.051607  0.050820  0.018933  0.023775
POA3008      0.034267  0.025472  0.009353  0.013202
POA3010      0.009353  0.001490  0.000435  0.000310
POA3011      0.036915  0.043019  0.024748  0.008422
POA3012      0.045937  0.041736  0.031866  0.010056
POA3013      0.018416  0.018416  0.014381  0.014754
POA3015      0.022658  0.020382  0.016037  0.017382
POA3016      0.025307  0.021189  0.013657  0.023651
POA3018      0.018271  0.018437  0.014236  0.006766
POA3019      0.023548  0.013098  0.012850  0.001200
POA3020      0.098578  0.073251  0.061808  0.004366
POA3021      0.131914  0.092308  0.077120  0.003787
POA3022      0.006601  0.005773  0.004221  0.000290
POA3023      0.139922  0.097999  0.079190  0.010884
POA3024      0.044509  0.027417  0.031080  0.003042
POA3025      0.020113  0.021789  0.013202  0.002876
POA3026      0.000000  0.000310  0.000062  0.000000
POA3027      0.015747  0.006125  0.007015  0.002835
POA3028      0.056428  0.038426  0.037246  0.004345
POA3029      0.203964  0.124361  0.122354  0.013284
POA3030      0.201543  0.117657  0.114863  0.036129
POA3031      0.023838  0.033584  0.017299  0.015043
POA3032      0.050696  0.043102  0.034908  0.023858
POA3033      0.023796  0.020589  0.016761  0.005670
POA3034      0.020672  0.018064  0.013450  0.003911
...               ...       ...       ...       ...
POA3950      0.005152  0.007284  0.007594  0.000993
POA3951      0.001407  0.001738  0.001324  0.000248
POA3953      0.009291  0.012953  0.010739  0.001821
POA3954      0.000621  0.000331  0.000828  0.000000
POA3956      0.003538  0.008629  0.003725  0.000766
POA3957      0.000352  0.000414  0.000435  0.000062
POA3958      0.000372  0.000559  0.000269  0.000062
POA3959      0.001262  0.002193  0.001697  0.000331
POA3960      0.002752  0.005566  0.003021  0.000476
POA3962      0.001076  0.001655  0.001386  0.000228
POA3964      0.000228  0.000000  0.000269  0.000000
POA3965      0.000372  0.000724  0.000248  0.000000
POA3966      0.000828  0.000538  0.001014  0.000124
POA3967      0.000062  0.000248  0.000124  0.000000
POA3971      0.005132  0.007925  0.005670  0.000973
POA3975      0.032487  0.018189  0.019740  0.002318
POA3976      0.061725  0.035674  0.040619  0.001428
POA3977      0.165415  0.116912  0.137914  0.014754
POA3978      0.017588  0.011381  0.015002  0.002255
POA3979      0.000393  0.000476  0.000621  0.000000
POA3980      0.004366  0.005214  0.004635  0.000600
POA3981      0.006746  0.009705  0.007760  0.000952
POA3984      0.006518  0.009994  0.007677  0.001014
POA3987      0.002462  0.002483  0.001945  0.000538
POA3988      0.001717  0.001573  0.001780  0.000228
POA3990      0.000103  0.000186  0.000290  0.000000
POA3991      0.000517  0.000621  0.000393  0.000062
POA3992      0.001242  0.001262  0.001697  0.000248
POA3995      0.010263  0.020175  0.015974  0.001924
POA3996      0.005359  0.009291  0.008070  0.002235

[698 rows x 4 columns]

# Inspect income distributions based on age groups
## Variable Three: Age group


In [216]:
vic_pa = vic_p.groupby(['age_low', 'income_low'])['number'].sum()
vic_p_y15_19 = vic_pa.xs(15, level='age_low')
vic_p_y20_24 = vic_pa.xs(20, level='age_low')
vic_p_y25_34 = vic_pa.xs(25, level='age_low')
vic_p_y35_44 = vic_pa.xs(35, level='age_low')
vic_p_y45_54 = vic_pa.xs(45, level='age_low')
vic_p_y55_64 = vic_pa.xs(55, level='age_low')
vic_p_y65_74 = vic_pa.xs(65, level='age_low')
vic_p_y75_84 = vic_pa.xs(75, level='age_low')
vic_p_y85_99 = vic_pa.xs(85, level='age_low')

vic_p_age_income_num = pd.concat([vic_p_y15_19, vic_p_y20_24, vic_p_y25_34, vic_p_y35_44, vic_p_y45_54, 
                                  vic_p_y55_64, vic_p_y65_74, vic_p_y75_84, vic_p_y85_99], 
                                  axis=1, 
                                  keys=['15-19', '20-24', '25-34', '35-44', '45-54', 
                                        '55-64', '65-74', '75-84', '85 over'])


vic_p_age_income_percent = vic_p_age_income_num / vic_p_pop * 100

In [217]:
#vic_pa = vic_p.groupby(['age_low', 'income_low'])['number'].sum()
#vic_pa

## Frequency Table: Income by Age Group (Count)

In [218]:
vic_p_age_income_num

15-19  20-24   25-34  35-44  45-54  55-64  65-74  75-84  85 over
income_low                                                                   
-99.0        32862  32492   70733  58703  57918  53491  50116  38723    26868
-1.0        165838  62796   68253  50995  45354  55922  28224  11623     4820
 1.0         84352  32658   19253  19303  17816  20824  14946   6313     2368
 150.0       32400  52283   46905  40254  40149  51665  48103  27930     8992
 300.0       11900  34667   39502  36449  41013  54240  98264  67681    22071
 400.0       10601  34130   43854  39152  45322  50979  79158  57280    29932
 500.0        8394  40461   61533  52135  54208  50940  52884  27122    12512
 650.0        4611  41922   80686  62343  63591  53286  36055  16407     6473
 800.0        2404  38724  106246  76172  76978  60394  28549  10073     4310
 1000.0        923  25508  117836  84040  81247  61804  24105   7506     3256
 1250.0        183   9335   84087  65889  58795  41001  13371   3318     1302
 1500.0        136   3646   59844  60213  53287  35602   9728   2588      978
 1750.0         47   1382   33451  44562  39192  25214   6255   1568      643
 2000.0         96   1181   41677  70360  58992  35007   9202   2469      943
 3000.0        262    914   13171  43584  45379  25999   9403   3070     1304

In [219]:
# sanity check
sum(vic_p_age_income_num.sum())

4832708

## Frequency Table: Income by Age Group

Percentage is calculated by using the whole population

In [220]:
vic_p_age_income_percent

15-19     20-24     25-34     35-44     45-54     55-64  \
income_low                                                               
-99.0       0.679991  0.672335  1.463631  1.214702  1.198459  1.106854   
-1.0        3.431575  1.299396  1.412314  1.055205  0.938480  1.157157   
 1.0        1.745440  0.675770  0.398389  0.399424  0.368655  0.430897   
 150.0      0.670432  1.081857  0.970574  0.832949  0.830776  1.069069   
 300.0      0.246239  0.717341  0.817389  0.754215  0.848655  1.122352   
 400.0      0.219359  0.706229  0.907442  0.810146  0.937818  1.054874   
 500.0      0.173691  0.837232  1.273261  1.078795  1.121690  1.054067   
 650.0      0.095412  0.867464  1.669582  1.290022  1.315846  1.102612   
 800.0      0.049744  0.801290  2.198478  1.576176  1.592854  1.249693   
 1000.0     0.019099  0.527820  2.438302  1.738984  1.681190  1.278869   
 1250.0     0.003787  0.193163  1.739956  1.363397  1.216606  0.848406   
 1500.0     0.002814  0.075444  1.238312  1.245947  1.102632  0.736688   
 1750.0     0.000973  0.028597  0.692179  0.922092  0.810974  0.521736   
 2000.0     0.001986  0.024438  0.862394  1.455913  1.220682  0.724376   
 3000.0     0.005421  0.018913  0.272539  0.901855  0.938997  0.537980   

               65-74     75-84   85 over  
income_low                                
-99.0       1.037017  0.801269  0.555962  
-1.0        0.584020  0.240507  0.099737  
 1.0        0.309268  0.130631  0.048999  
 150.0      0.995363  0.577937  0.186065  
 300.0      2.033311  1.400478  0.456700  
 400.0      1.637964  1.185257  0.619363  
 500.0      1.094293  0.561217  0.258902  
 650.0      0.746062  0.339499  0.133941  
 800.0      0.590745  0.208434  0.089184  
 1000.0     0.498789  0.155317  0.067374  
 1250.0     0.276677  0.068657  0.026941  
 1500.0     0.201295  0.053552  0.020237  
 1750.0     0.129431  0.032446  0.013305  
 2000.0     0.190411  0.051089  0.019513  
 3000.0     0.194570  0.063525  0.026983

In [221]:
vic_p_age_pop = vic_p.groupby(['age_low'])['number'].sum()
# index is needed for the divide() in the next cell
vic_p_age_pop.index =['15-19', '20-24', '25-34', '35-44', '45-54', '55-64', '65-74', '75-84', '85 over']
vic_p_age_pop

15-19      355009
20-24      412099
25-34      887031
35-44      804154
45-54      779241
55-64      676368
65-74      508363
75-84      283671
85 over    126772
Name: number, dtype: int64

## Frequency Table: Income by Age Group

Percentage is calculated by using each age-group population

There are different ways to investiagte the distriubtion of income. 
If I focus on the bigget group, we can see the biggest income group reaches its peak at \$1000 for the age 25 to 64.
Then it drops sharply afterwards. I think it's due to retirement.


In [226]:
vic_p_age_income_percent2 = vic_p_age_income_num.divide(vic_p_age_pop, axis=1) * 100
vic_p_age_income_percent2

15-19      20-24      25-34      35-44      45-54     55-64  \
income_low                                                                    
-99.0        9.256667   7.884513   7.974129   7.299970   7.432617  7.908565   
-1.0        46.713745  15.238086   7.694545   6.341447   5.820279  8.267984   
 1.0        23.760524   7.924795   2.170499   2.400411   2.286327  3.078797   
 150.0       9.126529  12.687000   5.287865   5.005758   5.152321  7.638593   
 300.0       3.352028   8.412299   4.453283   4.532590   5.263198  8.019303   
 400.0       2.986121   8.281990   4.943908   4.868719   5.816172  7.537169   
 500.0       2.364447   9.818272   6.936962   6.483211   6.956513  7.531403   
 650.0       1.298840  10.172798   9.096187   7.752620   8.160633  7.878256   
 800.0       0.677166   9.396771  11.977710   9.472315   9.878587  8.929163   
 1000.0      0.259993   6.189775  13.284316  10.450735  10.426428  9.137629   
 1250.0      0.051548   2.265232   9.479601   8.193580   7.545163  6.061937   
 1500.0      0.038309   0.884739   6.746551   7.487745   6.838321  5.263703   
 1750.0      0.013239   0.335356   3.771120   5.541476   5.029509  3.727852   
 2000.0      0.027042   0.286582   4.698483   8.749568   7.570444  5.175733   
 3000.0      0.073801   0.221791   1.484841   5.419857   5.823487  3.843913   

                65-74      75-84    85 over  
income_low                                   
-99.0        9.858310  13.650673  21.193955  
-1.0         5.551938   4.097352   3.802101  
 1.0         2.940025   2.225465   1.867920  
 150.0       9.462333   9.845913   7.093049  
 300.0      19.329495  23.858977  17.409996  
 400.0      15.571157  20.192406  23.610892  
 500.0      10.402803   9.561076   9.869687  
 650.0       7.092373   5.783813   5.106017  
 800.0       5.615869   3.550945   3.399804  
 1000.0      4.741690   2.646023   2.568390  
 1250.0      2.630207   1.169665   1.027041  
 1500.0      1.913593   0.912324   0.771464  
 1750.0      1.230420   0.552753   0.507210  
 2000.0      1.810124   0.870374   0.743855  
 3000.0      1.849663   1.082240   1.028618

# Review

It's a long process to produce usable data as seen shown in another jupyter notebook.

I convert the following data to numbers to make it more agreeable with pandas:
 * age: over 85 -> 99
 * negative income -> -1
 * zero income -> 0
 * unknown income -> -99
 * income more than 3000 -> 9999

These choices may seem easy to handle at the moment but I expect this may not be the case for some questions and I may need to modify them accordingly. For example, calculating the mean income of a specific group can be an issue. 

The postal data is long and it would be good to include and/or show geo-location information. I'm not sure if this is part of the curriculum.

## The frequency based on gender, post code and age
The first table (gender) is fairly simple and we can even draw a quick conclusion from it without too much inspection.

The second tables are very long (698) due to the large number of postal areas. It is a lot harder to draw any quick conclusion and I expect to use other tools to help with it later.

The third table is not as simple as the first table and besides quick investiagtion, further detailed exploration is definitely required, particularly if I want to answer my question related to gender.

a few sentences describing your frequency distributions in terms of the values the variables take, how often they take them, the presence of missing data, etc.